### PR TITLE
Defending against past corrupt builds

### PR DESCRIPTION
**Pull request by Vlatombe**
cc @jglick @lemeurherve 

I've seen builds failing in the wild during `junit` step execution with an exception like

```
java.lang.NullPointerException: Cannot invoke "hudson.model.Run.getRootDir()" because "this.run" is null
	at PluginClassLoader for junit//hudson.tasks.junit.TestResultAction.getDataFilePath(TestResultAction.java:154)
	at PluginClassLoader for junit//hudson.tasks.junit.TestResultAction.loadCached(TestResultAction.java:317)
	at PluginClassLoader for junit//hudson.tasks.junit.TestResultAction.load(TestResultAction.java:279)
	at PluginClassLoader for junit//hudson.tasks.junit.TestResultAction.getResult(TestResultAction.java:166)
	at PluginClassLoader for junit//hudson.tasks.junit.TestResultAction.getResult(TestResultAction.java:70)
	at PluginClassLoader for junit//hudson.tasks.test.AbstractTestResultAction.findCorrespondingResult(AbstractTestResultAction.java:286)
	at PluginClassLoader for junit//hudson.tasks.test.TestResult.getPreviousResult(TestResult.java:146)
	at PluginClassLoader for junit//hudson.tasks.junit.TestResult.getPreviousResult(TestResult.java:467)
	at PluginClassLoader for junit//hudson.tasks.junit.CaseResult.getPreviousResult(CaseResult.java:768)
	at PluginClassLoader for junit//hudson.tasks.junit.CaseResult.recomputeFailedSinceIfNeeded(CaseResult.java:669)
	at PluginClassLoader for junit//hudson.tasks.junit.CaseResult.freeze(CaseResult.java:950)
	at PluginClassLoader for junit//hudson.tasks.junit.SuiteResult.freeze(SuiteResult.java:730)
	at PluginClassLoader for junit//hudson.tasks.junit.TestResult.freeze(TestResult.java:1119)
	at PluginClassLoader for junit//hudson.tasks.junit.TestResultAction.setResult(TestResultAction.java:125)
	at PluginClassLoader for junit//hudson.tasks.junit.TestResultAction.<init>(TestResultAction.java:96)
	at PluginClassLoader for junit//hudson.tasks.junit.JUnitResultArchiver.parseAndSummarize(JUnitResultArchiver.java:301)
	at PluginClassLoader for junit//hudson.tasks.junit.pipeline.JUnitResultsStepExecution.run(JUnitResultsStepExecution.java:62)
	at PluginClassLoader for junit//hudson.tasks.junit.pipeline.JUnitResultsStepExecution.run(JUnitResultsStepExecution.java:27)
	at PluginClassLoader for workflow-step-api//org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

While the root cause for such corruption is still to be identified, the plugin should defend against that condition and ignore such a build.

Similar to https://github.com/jenkinsci/parallel-test-executor-plugin/pull/282

### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/hudson/tasks/test/TestResult.java [hunk 1]
```diff
@@ -132,7 +136,8 @@ public Result getBuildResult() {
      */
     @Override
     public TestResult getPreviousResult() {
-        Run<?, ?> b = getRun();
+        Run<?, ?> originBuild = getRun();
+        Run<?, ?> b = originBuild;
```
Comment (timja) [inline src/main/java/hudson/tasks/test/TestResult.java:140]: why do you need both variables?

Comment (Vlatombe) [reply src/main/java/hudson/tasks/test/TestResult.java]: Needed it to look up job (as I can't really trust past builds here), but I just reworked it in 254298b

File: src/main/java/hudson/tasks/test/TestResult.java [hunk 2]
```diff
@@ -136,17 +140,27 @@ public TestResult getPreviousResult() {
         if (b == null) {
             return null;
         }
+        Job<?, ?> job = b.getParent();
         while (true) {
             b = b.getPreviousBuild();
             if (b == null) {
                 return null;
             }
-            AbstractTestResultAction r = b.getAction(getParentAction().getClass());
```
Comment (Vlatombe) [inline src/main/java/hudson/tasks/test/TestResult.java:144]: (ignore ws)

File: src/main/java/hudson/tasks/test/TestResult.java (hunk 1 above)
Comment (timja) [reply src/main/java/hudson/tasks/test/TestResult.java]: much better, thanks

**Review by timja — APPROVED** [review]
Thanks!

**Comment by Vlatombe** [comment]
Thanks for the fast turnaround!

File: src/main/java/hudson/tasks/test/TestResult.java (hunk 2 above)
Comment (Lomasuno) [reply src/main/java/hudson/tasks/test/TestResult.java:144]: Hey ocupo que me ayudes con un numero

Comment (Lomasuno) [reply src/main/java/hudson/tasks/test/TestResult.java:144]: Your's ignoret

**Pull request merged**